app/javascript/elm/tests/MainTests.elm

```diff
@@ -525,132 +528,173 @@ sessionWithTitle title =
+                        }
 
                     expected =
-                        { model | selectedSessionId = Just id }
+                        List.map (\x -> Slc.containing [ Slc.text x ]) [ title, username, sensorName ]
                 in
-                model
-                    |> update (ToggleSessionSelection id)
-                    |> Tuple.first
-                    |> Expect.equal expected
-        , fuzz int "when id was selected ToggleSessionSelection deselects it" <|
-            \id ->
+                { defaultModel | selectedSession = Success selectedSession_ }
+                    |> view
+                    |> Query.fromHtml
+                    |> Query.has [ Slc.all expected ]
+        , fuzz3 float float float "with selection the session average min and max are shown" <|
+            \average min max ->
                 let
-                    model =
-                        { defaultModel | selectedSessionId = Just id }
+                    selectedSession_ =
+                        { selectedSession
+                            | average = average
+                            , min = min
+                            , max = max
+                        }
 
                     expected =
-                        { model | selectedSessionId = Nothing }
+                        List.map (\x -> Slc.containing [ Slc.text <| String.fromFloat x ]) [ average, min, max ]
                 in
-                model
-                    |> update (ToggleSessionSelection id)
-                    |> Tuple.first
-                    |> Expect.equal expected
-        , fuzz int "when another id was selected ToggleSessionSelection selects the new one" <|
-            \id ->
+                { defaultModel | selectedSession = Success selectedSession_ }
+                    |> view
+                    |> Query.fromHtml
+                    |> Query.has [ Slc.all expected ]
+        , fuzz string "with selection the session timeRange is shown" <|
+            \timeRange ->
                 let
-                    model =
-                        { defaultModel | selectedSessionId = Just (id + 1) }
+                    selectedSession_ =
+                        { selectedSession | timeRange = timeRange }
 
                     expected =
-                        { model | selectedSessionId = Just id }
+                        List.map (\x -> Slc.containing [ Slc.text x ]) [ timeRange ]
                 in
-                model
-                    |> update (ToggleSessionSelection id)
-                    |> Tuple.first
-                    |> Expect.equal expected
-        , fuzz int "with no selections ToggleSessionSelection tells javascript what was selected" <|
+                { defaultModel | selectedSession = Success selectedSession_ }
+                    |> view
+                    |> Query.fromHtml
+                    |> Query.has [ Slc.all expected ]
+        ]
+
+
+updateTests : Test
+updateTests =
```

Comment:
I think I kinda like this way of organizing tests better, will try that out next time

Reply:
I'm not sure how much I like it. It's risky to forget about exposing (and running) some tests